test/metabase/query_processor/middleware/optimize_temporal_filters_test.clj

```diff
@@ -214,3 +214,140 @@
                                (t/zoned-date-time "2019-10-02T00:00Z[UTC]")]}
            (mt/$ids checkins
              (filter->sql [:between !day.date "2019-09-01" "2019-10-01"]))))))
+
+(deftest optimize-relative-datetimes-test
```

Comment:
I'm just noticing that everything here is operating on a `:month` resolution (as per the #11837 starter comment).  But does it make sense to also cover other resolutions, like `:hour`?  Or does `validate-temporal-bucketing-test` cover that same path?

Reply:
that's a good point. I'll update the tests to cover all the supported units